hokusai/development.yml

```diff
@@ -2,24 +2,39 @@
 version: '2'
 services:
   rosalind:
-    command: sh -c 'bundle exec rails db:setup && bundle exec puma -C config/puma.config'
+    command: sh -c 'bundle exec rails db:migrate && bundle exec puma -C config/puma.config'
     env_file: ../.env
     extends:
       file: build.yml
       service: rosalind
     ports:
-      - 3000:3000
+      - 5000:5000
     volumes:
       - ../:/app
     depends_on:
       - rosalind-postgres
+      - rosalind-redis
     environment:
       - DATABASE_URL=postgresql://postgres:@rosalind-postgres/rosalind_development
       - RAILS_ENV=development
       - REDIS_URL=redis://rosalind-redis
+  worker:
+    extends:
+      file: build.yml
+      service: rosalind
+    environment:
+      - RAILS_ENV=development
+      - DATABASE_URL=postgresql://postgres:@rosalind-postgres/rosalind_development
+      - REDIS_URL=redis://rosalind-redis
+    env_file: ../.env
+    command: bundle exec sidekiq
```

Comment:
Sidekiq not quite working yet... 

Jobs do get enqueued but then seem to fail with an error: 
```
NameError: uninitialized constant ProcessBatchUpdateJob
```

<details>
<summary>full stack trace</summary>
<pre>
rosalind_1           | [ActiveJob] Enqueued ProcessBatchUpdateJob (Job ID: 15c16025-4586-4ecb-b8a0-63a979f4b0e0) to Sidekiq(default) with arguments: 2
worker_1             | 2019-06-19T14:20:26.672Z 6 TID-gsc46831u ProcessBatchUpdateJob JID-1a48f9dc39bd1a27d999b7b7 INFO: start
worker_1             | 2019-06-19T14:20:26.675Z 6 TID-gsc46831u ProcessBatchUpdateJob JID-1a48f9dc39bd1a27d999b7b7 INFO: fail: 0.003 sec
worker_1             | 2019-06-19T14:20:26.675Z 6 TID-gsc46831u WARN: {"context":"Job raised exception","job":{"class":"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper","wrapped":"ProcessBatchUpdateJob","queue":"default","args":[{"job_class":"ProcessBatchUpdateJob","job_id":"15c16025-4586-4ecb-b8a0-63a979f4b0e0","provider_job_id":null,"queue_name":"default","priority":null,"arguments":[2],"executions":0,"locale":"en"}],"retry":true,"jid":"1a48f9dc39bd1a27d999b7b7","created_at":1560954026.67109,"enqueued_at":1560954026.6711452},"jobstr":"{\"class\":\"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper\",\"wrapped\":\"ProcessBatchUpdateJob\",\"queue\":\"default\",\"args\":[{\"job_class\":\"ProcessBatchUpdateJob\",\"job_id\":\"15c16025-4586-4ecb-b8a0-63a979f4b0e0\",\"provider_job_id\":null,\"queue_name\":\"default\",\"priority\":null,\"arguments\":[2],\"executions\":0,\"locale\":\"en\"}],\"retry\":true,\"jid\":\"1a48f9dc39bd1a27d999b7b7\",\"created_at\":1560954026.67109,\"enqueued_at\":1560954026.6711452}"}
worker_1             | 2019-06-19T14:20:26.675Z 6 TID-gsc46831u WARN: NameError: uninitialized constant ProcessBatchUpdateJob
worker_1             | 2019-06-19T14:20:26.676Z 6 TID-gsc46831u WARN: /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/inflector/methods.rb:283:in `const_get'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/inflector/methods.rb:283:in `block in constantize'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/inflector/methods.rb:281:in `each'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/inflector/methods.rb:281:in `inject'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/inflector/methods.rb:281:in `constantize'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/core_ext/string/inflections.rb:68:in `constantize'
worker_1             | /usr/local/bundle/gems/activejob-5.2.1/lib/active_job/core.rb:41:in `deserialize'
worker_1             | /usr/local/bundle/gems/activejob-5.2.1/lib/active_job/execution.rb:23:in `block in execute'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/callbacks.rb:109:in `block in run_callbacks'
worker_1             | /usr/local/bundle/gems/activejob-5.2.1/lib/active_job/railtie.rb:28:in `block (4 levels) in <class:Railtie>'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/execution_wrapper.rb:87:in `wrap'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/reloader.rb:73:in `block in wrap'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/execution_wrapper.rb:83:in `wrap'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/reloader.rb:72:in `wrap'
worker_1             | /usr/local/bundle/gems/activejob-5.2.1/lib/active_job/railtie.rb:27:in `block (3 levels) in <class:Railtie>'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/callbacks.rb:118:in `instance_exec'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/callbacks.rb:118:in `block in run_callbacks'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/callbacks.rb:136:in `run_callbacks'
worker_1             | /usr/local/bundle/gems/activejob-5.2.1/lib/active_job/execution.rb:22:in `execute'
worker_1             | /usr/local/bundle/gems/activejob-5.2.1/lib/active_job/queue_adapters/sidekiq_adapter.rb:42:in `perform'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/processor.rb:185:in `execute_job'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/processor.rb:167:in `block (2 levels) in process'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/middleware/chain.rb:128:in `block in invoke'
worker_1             | /usr/local/bundle/gems/ddtrace-0.17.2/lib/ddtrace/contrib/sidekiq/tracer.rb:37:in `block in call'
worker_1             | /usr/local/bundle/gems/ddtrace-0.17.2/lib/ddtrace/tracer.rb:296:in `trace'
worker_1             | /usr/local/bundle/gems/ddtrace-0.17.2/lib/ddtrace/contrib/sidekiq/tracer.rb:29:in `call'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/middleware/chain.rb:130:in `block in invoke'
worker_1             | /usr/local/bundle/gems/sentry-raven-2.7.4/lib/raven/integrations/sidekiq.rb:9:in `call'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/middleware/chain.rb:130:in `block in invoke'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/middleware/chain.rb:133:in `invoke'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/processor.rb:166:in `block in process'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/processor.rb:137:in `block (6 levels) in dispatch'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/job_retry.rb:108:in `local'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/processor.rb:136:in `block (5 levels) in dispatch'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/rails.rb:42:in `block in call'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/execution_wrapper.rb:87:in `wrap'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/reloader.rb:73:in `block in wrap'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/execution_wrapper.rb:87:in `wrap'
worker_1             | /usr/local/bundle/gems/activesupport-5.2.1/lib/active_support/reloader.rb:72:in `wrap'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/rails.rb:41:in `call'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/processor.rb:132:in `block (4 levels) in dispatch'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/processor.rb:243:in `stats'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/processor.rb:127:in `block (3 levels) in dispatch'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/job_logger.rb:8:in `call'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/processor.rb:126:in `block (2 levels) in dispatch'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/job_retry.rb:73:in `global'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/processor.rb:125:in `block in dispatch'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/logging.rb:48:in `with_context'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/logging.rb:42:in `with_job_hash_context'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/processor.rb:124:in `dispatch'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/processor.rb:165:in `process'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/processor.rb:83:in `process_one'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/processor.rb:71:in `run'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/util.rb:16:in `watchdog'
worker_1             | /usr/local/bundle/gems/sidekiq-5.2.3/lib/sidekiq/util.rb:25:in `block in safe_thread'
</pre>
</details>

Reply:
We may need to add another container for this, some apps run sidekiq in one -- here's an example using this: https://github.com/artsy/reflection/blob/master/hokusai/staging.yml#L113

Reply:
Ok — I didn't see any obvious problems with the way the `worker` service was defined in `development.yml`

So after a certain amount of table-flipping (`docker system prune --all` etc) I tried again, and now it seems to work fine 🎉 

So that covers the local `hokusai dev start` case.

Still need to set everything up in the stage and prod environments, and would love some help with that if you're able @eessex.